plugin/server.go

```diff
@@ -61,6 +73,41 @@ func getTCPListener(minPort, maxPort int) (net.Listener, error) {
 	return nil, fmt.Errorf(`no available port in the range %d to %d`, minPort, maxPort)
 }
 
+var tempFileAttempts = 10
+
+// tempFile generates random file name in a given directory
+// file name has a form of <plugin-name><random-string>.socket
+// the function returns an error is given directory doesn't exist.
+func tempFile(dir, prefix string) (string, error) {
```

Comment:
This isn't safe from a concurrency perspective.
Consider the following:
- Plugin A is started from two different processes simultaneously.
- Both call tempFile before the other has created the actual socket.

This means that both will then make an attempt to use a socket named "\<dir\>/A.socket"

A safer way, which would also avoid the need to use a loop and random numbers, would be to add the process PID to the socket name. I.e.
```
return prefix + "-" + strconv.Itoa(os.Getpid()) + ".socket"
```